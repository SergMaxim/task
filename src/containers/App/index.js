// Core
import React, { Component } from 'react';

// Instruments
import Styles from './styles.scss';
import { string } from 'prop-types';

//import Catcher from '../../components/Catcher';
import Emails from '../../components/Emails';
import Head from '../../components/Head';

const option = {
    firstName: 'Максим',
    lastName:  'Сергийчук'
};

export default class App extends Component {
    static childContextTypes = {
        firstName: string.isRequired,
        lastName:  string.isRequired
    };

    getChildContext () {
        return option;
    }
    render () {
        return (
            <section className = { Styles.app }>
                <Head />
                <Emails />
            </section>
        );
    }
}
