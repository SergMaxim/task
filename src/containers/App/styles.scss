@import '../../theme/palette';
@import url('https://fonts.googleapis.com/css?family=Poppins');

.app {
  font-family: 'Poppins', sans-serif;
}
