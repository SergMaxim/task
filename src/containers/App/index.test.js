import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import dom from 'react-test-renderer';

// Instruments
import App from './index';

configure({ adapter: new Adapter() });

global.fetch = jest.fn(() => ({
    data: [
        {
            name:  'Mansion Tech',
            theme: 'New Tasks',
            body:  'New Tasks',
            star:  true,
            type:  1,
            time:  1513545795
        },
        {
            name:  'Mansion Tech',
            theme: 'New Tasks',
            body:  'New Tasks',
            star:  true,
            type:  1,
            time:  1513545795
        }
    ],
    message: 'the request has succeeded'
}));

const result = mount(<App />);
const renderTree = dom.create(<App />).toJSON();

describe('App component', () => {
    test('App component should correspond to its snapshot counterpart', () => {
        expect(renderTree).toMatchSnapshot();
    });
});
