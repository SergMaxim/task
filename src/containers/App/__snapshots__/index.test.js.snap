// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component App component should correspond to its snapshot counterpart 1`] = `
<section
  className="app"
>
  <section
    className="head"
  >
    <div
      className="bell"
    >
      <span
        className="count"
      >
        2
      </span>
    </div>
    <div
      className="letter"
    >
      <span
        className="count"
      >
        2
      </span>
    </div>
    <div
      className="star"
    />
    <div
      className="user"
    >
      <img
        src="user.png"
      />
      <div>
        <span
          onClick={[Function]}
        >
          Максим
           
          Сергийчук
        </span>
        <ul />
      </div>
    </div>
    <form
      className="search"
    >
      <input
        placeholder="search"
        type="text"
      />
      <input
        type="submit"
        value=""
      />
    </form>
  </section>
  <section
    className="emails"
  >
    <section
      className="header"
    >
      <ul>
        <li>
          <a>
            Home
          </a>
        </li>
        <li>
          Email
        </li>
      </ul>
      <p>
        Email
      </p>
    </section>
    <div
      className="emailsBody"
    >
      <section
        className="menu"
      >
        <div
          className="button"
        >
          Compose Mail
        </div>
      </section>
      <section
        className="letters"
      >
        <section
          className="header"
        >
          <div>
            Inbox 
            <span>
              0
            </span>
          </div>
          <form
            className="search"
            onSubmit={[Function]}
          >
            <input
              onKeyUp={[Function]}
              placeholder="search"
              type="text"
            />
            <input
              type="submit"
              value=""
            />
          </form>
        </section>
      </section>
    </div>
  </section>
</section>
`;
